omnigibson/systems/macro_particle_system.py

```diff
@@ -359,6 +363,33 @@ def initialize(cls):
         # Initialize mutable class variables so they don't automatically get overridden by children classes
         cls._particles_info = dict()
         cls._particles_local_mat = dict()
+        cls._cloth_face_ids = dict()
+
+    @classmethod
+    def update(cls):
+        # Run super first
+        super().update()
+
+        # Iterate over all objects, and update all particles belonging to any cloth objects
+        for name, obj in cls._group_objects.items():
+            if obj.prim_type == PrimType.CLOTH:
+                # Update the transforms
+                group = cls.get_group_name(obj=obj)
+                cloth = obj.root_link
+                face_ids = cls._cloth_face_ids[group]
+                normals = cloth.compute_face_normals(face_ids=face_ids)
+                positions = cloth.particle_positions[cloth.faces[face_ids]].mean(axis=1)
+                # Orientations are the normals
+                z_up = np.zeros_like(normals)
+                z_up[:, 2] = 1.0
+                orientations = T.axisangle2quat(T.vecs2axisangle(z_up, normals))
+                if cls._CLIP_INTO_OBJECTS:
+                    offset = np.array([cls._particle_object.aabb_extent * particle.scale for particle in cls._group_particles[group].values()]) / 2.0
+                    # Shift the particles halfway down
+                    positions -= normals * offset[2].reshape(-1, 1)
```

Comment:
Not sure if this is still needed. Actually, the behavior is different from the rigid body case. Without additional shifting, the macro particle is already (half) clip into the cloth objects. 

Maybe it should be the following instead (I am not sure, happy to discuss!):
```
                if not cls._CLIP_INTO_OBJECTS:
                    offset = np.array([cls._particle_object.aabb_extent * particle.scale for particle in cls._group_particles[group].values()]) / 2.0
                    # Shift the particles halfway down
                    positions += normals * offset[2].reshape(-1, 1)
```

Reply:
done!